action/protocol/producer/reward.go

```diff
@@ -0,0 +1,208 @@
+// Copyright (c) 2019 IoTeX
+// This is an alpha (internal) release and is not suitable for production. This source code is provided 'as is' and no
+// warranties are given as to title or non-infringement, merchantability or fitness for purpose and, to the extent
+// permitted by law, all liability for your use of the code is disclaimed. This source code is governed by Apache
+// License 2.0 that can be found in the LICENSE file.
+
+package producer
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/pkg/errors"
+
+	"github.com/iotexproject/iotex-core/action/protocol"
+	"github.com/iotexproject/iotex-core/action/protocol/producer/producerpb"
+	"github.com/iotexproject/iotex-core/address"
+	"github.com/iotexproject/iotex-core/pkg/enc"
+	"github.com/iotexproject/iotex-core/state"
+)
+
+type rewardHistory struct{}
+
+// Serialize serializes reward history state into bytes
+func (b rewardHistory) Serialize() ([]byte, error) {
+	gen := producerpb.RewardHistory{}
+	return proto.Marshal(&gen)
+}
+
+// Deserialize deserializes bytes into reward history state
+func (b *rewardHistory) Deserialize(data []byte) error {
+	gen := producerpb.RewardHistory{}
+	if err := proto.Unmarshal(data, &gen); err != nil {
+		return err
+	}
+	return nil
+}
+
+type rewardAccount struct {
+	balance *big.Int
+}
+
+// Serialize serializes account state into bytes
+func (a rewardAccount) Serialize() ([]byte, error) {
+	gen := producerpb.Account{
+		Balance: a.balance.Bytes(),
+	}
+	return proto.Marshal(&gen)
+}
+
+// Deserialize deserializes bytes into account state
+func (a *rewardAccount) Deserialize(data []byte) error {
+	gen := producerpb.Account{}
+	if err := proto.Unmarshal(data, &gen); err != nil {
+		return err
+	}
+	a.balance = big.NewInt(0).SetBytes(gen.Balance)
+	return nil
+}
+
+// GrantReward grants the block reward (token) to the block producer
+func (p *Protocol) GrantBlockReward(
```

Comment:
@raullenchai 

I changed `SettleBlockReward` to `GrantBlockReward`. Hopefully it sounds more accurate.

Reply:
Yes, it does.